src/core/pipeline/pipeline.js

```diff
@@ -0,0 +1,35 @@
+export default class Pipeline {
```

Comment:
1. What if exception happens in user-defined stage? We should catch that stuff and stop the pipeline i guess.

2 and 3 - Probably it's not directly middleware's work. But probably we need a way to stop the middleware outside?